### PR TITLE
Fix #192

### LINKED ISSUE #192: Add a systemProperties.adoc with a table of system properties

### PR DESCRIPTION
**Pull request by j4sm1ne96**
add systemProperties.adoc

### CODE REVIEW

File: systemProperties.adoc [hunk 1]
```diff
@@ -0,0 +1,22 @@
+== System Properties
+Below, a number of relevant System Properties are listed.
+
+.System properties
+[cols=4*, options="header"]
+|===
+| Property Key | Default | Description | Java Variable Name (Type)
+| chronicle.network.callOnMissedHeartbeatOnDisconnect |`false` | If 'true', calls HeartBeatListener when we do not receive a heartbeat | _CALL_MISSED_HEARTBEAT_ON_DISCONNECT_ (boolean)
```
Comment (minborg) [inline systemProperties.adoc]: If `true` or set to "", ....

File: systemProperties.adoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+| connectionStrategy.pause.max.secs | 5 | The maximum amount of time between two events | _maxPauseSec_ (long)
+| connectionStrategy.pause.min.secs | 5 | The minimum amount of time between two events | _minPauseSec_ (long)
+| connectionStrategy.socketConnectionTimeoutMs | 1 | Minimum time before message displayed: "Timed out attempting to connect to " + socketAddress | socketConnectionTimeoutMs (int)
+| disable.tcp.nodelay | `false` | If 'true', displays message 'tcpNoDelay disabled' | _DISABLE_TCP_NODELAY_ (boolean)
+| ssl.test.payload | `false` | If 'true', returns testDataConnection | _SEND_DATA_BEFORE_SSL_HANDSHAKE_ (boolean)
+| tcp.client.buffer.size | 0 | Gets or sets the suggested buffer size | _tcpBufferSize_ (int)
+| TcpEventHandler.targetWriteSize | 1024 | Used to check if ByteBuffer did write something | _TARGET_WRITE_SIZE_ (int)
+| tcp.event.monitor.secs | 10 | Sets the interval in between polls | _MONITOR_POLL_EVERY_SEC_ (int)
+| tcp.failover.time | 2_000 | Amount of time before the failover is used | _failoverTimeout_ (long)
```
Comment (minborg) [inline systemProperties.adoc]: Please add unit of measurement (ms) when not apparent from the variable name (e.g. SEC, MS)

File: systemProperties.adoc (hunk 1 above)
Comment (JerryShea) [reply systemProperties.adoc]: I don't think we should, let's just say if set

**Review by minborg — APPROVED** [review]

File: systemProperties.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+== System Properties
+Below, a number of relevant System Properties are listed.
+
+
+NOTE: With `getBoolean`, returns if a System Property with the provided {@code systemPropertyKey} either exists, is set to "yes" or is set to "true", see
```
Comment (JerryShea) [inline systemProperties.adoc]: the @code doesn't work in adoc - see https://github.com/OpenHFT/Chronicle-Network/blob/Issue_192/systemProperties.adoc

File: systemProperties.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+https://github.com/OpenHFT/Chronicle-Core/blob/351e79ed593fa656c21b4e5a540a3a5831cd06a3/src/main/java/net/openhft/chronicle/core/Jvm.java#L1184[javadoc].
+
+.System properties
+[cols=4*, options="header"]
+|===
+| Property Key | Default | Description | Java Variable Name (Type)
+| chronicle.network.callOnMissedHeartbeatOnDisconnect |`false` | Setting this property to "yes", "true" or "" will enable this feature and calls HeartBeatListener when we do not receive a heartbeat | _CALL_MISSED_HEARTBEAT_ON_DISCONNECT_ (boolean)
```
Comment (JerryShea) [inline systemProperties.adoc]: Do we really need 'Setting this property to "yes", "true" or ""'?

**Comment by j4sm1ne96** [comment]
@JerryShea I have edited this and made changes in the table

**Review by JerryShea — DISMISSED** [review]
DM'd you

**Comment by j4sm1ne96** [comment]
@rogersimmons @JerryShea is this now approved or please let me know if there any suggestions?

**Review by rogersimmons — APPROVED** [review]

File: systemProperties.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+Below, a number of relevant System Properties are listed.
+
+
+NOTE: All boolean properties below are read using link:https://javadoc.io/static/net.openhft/chronicle-core/2.23ea13/net/openhft/chronicle/core/Jvm.html#getBoolean-java.lang.String-[net.openhft.chronicle.core.Jvm.getBoolean(java.lang.String)], and so are enabled if either `-Dflag` or `-Dflag=true` or `-Dflag=yes`
+
+.System properties
+[cols=4*, options="header"]
+|===
+| Property Key | Default | Description | Java Variable Name (Type)
+| chronicle.network.callOnMissedHeartbeatOnDisconnect |`false` | If enabled, calls HeartBeatListener when we do not receive a heartbeat | _CALL_MISSED_HEARTBEAT_ON_DISCONNECT_ (boolean)
```
Comment (nicktindall) [inline systemProperties.adoc:11]: actually, when set to true this calls `onMissedHeartbeat` when the connection is abruptly dropped. For some reason that behaviour was removed and then it was added back in behind this feature switch. There's no context/information on [the issue](https://github.com/OpenHFT/Chronicle-Network/issues/112) so I have no idea why.

Comment (nicktindall) [reply systemProperties.adoc:11]: i.e. normally `onMissedHeartbeat` would only be called when a connection had gone quiet for `heartbeatTimeoutMs`, it allows the implementer to return `true` to wait a bit longer or `false` to drop the idle connection immediately, but when this property is set to true, we will call `onMissedHeartbeat` when we detect a dropped connection, regardless of whether a timeout has occurred. The return value is ignored in that scenario.

File: systemProperties.adoc [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+| client.timeout | 500 | Number of milliseconds after which a client is timed out if they cannot to a host port | _pausePeriodMs_ (int)
```
Comment (nicktindall) [inline systemProperties.adoc:12]: > if they cannot to a host port

typo, I think it should be "if they cannot connect to a host port"

File: systemProperties.adoc [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+| connectionStrategy.pauseMillisBeforeReconnect | 500 | Allows control of a backoff strategy, and returns how long in milliseconds to pause before attempting reconnect | pauseMillisBeforeReconnect (long)
+| connectionStrategy.pause.max.secs | 5 | Controls how frequently the ConnectionStrategy.connect method is called | _maxPauseSec_ (long)
+| connectionStrategy.pause.min.secs | 5 | Controls how frequently the ConnectionStrategy.connect method is called | _minPauseSec_ (long)
```
Comment (nicktindall) [inline systemProperties.adoc:15]: These two (`pause.min.secs`/`pause.max.secs`) don't actually seem to be used? Unless there are ConnectionStrategies  in other repos that make use of them. Perhaps we could deprecate?

Comment (nicktindall) [reply systemProperties.adoc:15]: Ahh they're used in FIX, I wonder if they should be moved there? but it looks like they were only just added to network

See the PR where they were added https://github.com/OpenHFT/Chronicle-Network/pull/181
See the issue it was to resolve https://github.com/ChronicleEnterprise/Chronicle-FIX/issues/711

File: systemProperties.adoc [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+| connectionStrategy.socketConnectionTimeoutMs | 1 | Minimum time before message displayed: "Timed out attempting to connect to " + socketAddress | socketConnectionTimeoutMs (int)
```
Comment (nicktindall) [inline systemProperties.adoc:16]: So this is the timeout for one single attempt at connection (how long we wait for a socket to open). `client.timeout` seems to be the overall timeout (how long we keep retrying for).

i.e. if the strategy is to retry every X seconds, each individual attempt will timeout after `connectionStrategy.socketConnectionTimeoutMs` and we will stop retrying after `client.timeout`. It's probably worth making that clear the distinction between the two timeouts?

File: systemProperties.adoc [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+| disable.tcp.nodelay | `false` | If enabled, displays message 'tcpNoDelay disabled' and allows sending of successive small packets on the network | _DISABLE_TCP_NODELAY_ (boolean)
```
Comment (nicktindall) [inline systemProperties.adoc:17]: This is to disable https://en.wikipedia.org/wiki/Nagle%27s_algorithm for connections. When true, the connection will optimise for low packet latency at the cost of greater overall traffic, might be worth linking the wikipedia article as there's lots of caveats about the effect of disabling it.

File: systemProperties.adoc [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+| ssl.test.payload | `false` | If enabled, triggers testDataConnection to encode ByteBuffer message while there is at least one remaining element in the ByteBuffer | _SEND_DATA_BEFORE_SSL_HANDSHAKE_ (boolean)
```
Comment (nicktindall) [inline systemProperties.adoc:18]: This one seems to be purely used in an Integration test, I suggest we omit it

File: systemProperties.adoc [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+| tcp.client.buffer.size | 0 | Gets or sets the tcp buffer size | _tcpBufferSize_ (int)
```
Comment (nicktindall) [inline systemProperties.adoc:19]: This actually defaults to 4 MiB (kind of, it's a long story)

Comment (nicktindall) [reply systemProperties.adoc:19]: it sets the value for `SO_RCVBUF` (see https://docs.oracle.com/javase/7/docs/api/java/net/ServerSocket.html#setReceiveBufferSize(int))

File: systemProperties.adoc [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+| TcpEventHandler.targetWriteSize | 1024 | Determines the size of what is trying to be written to the ByteBuffer to check if ByteBuffer has capacity | _TARGET_WRITE_SIZE_ (int)
+| tcp.event.monitor.secs | 10 | Sets the interval in between polls | _MONITOR_POLL_EVERY_SEC_ (int)
```
Comment (nicktindall) [inline systemProperties.adoc:21]: This is how often the network stats listener will be notified of the network statistics (e.g. writeBps/readBps/socketPollCountPerSecond), and the window over which they are measured.

**Pull request merged**